config/unix-xl.cmake

```diff
@@ -22,14 +22,57 @@ query_openmp_availability()
 if( NOT CXX_FLAGS_INITIALIZED )
    set( CXX_FLAGS_INITIALIZED "yes" CACHE INTERNAL "using draco settings." )
 
-  # On Darwin, we also need this config file:
-  # -F/projects/opt/ppc64le/ibm/xlc-16.1.1.2/xlC/16.1.1/etc/xlc.cfg.rhel.7.5.gcc.7.3.0.cuda.9.2
-  # -qfloat=nomaf -qxlcompatmacros
-  set( CMAKE_C_FLAGS                "-g --gcc-toolchain=/usr/tce/packages/gcc/gcc-8.3.1" ) # -qarch=auto
+  set( CMAKE_C_FLAGS             "-g" )
+  if( EXISTS /usr/gapps )
+    # ATS-2
+    string( APPEND CMAKE_C_FLAGS " --gcc-toolchain=/usr/tce/packages/gcc/gcc-8.3.1" )
+  elseif(EXISTS /projects/opt/ppc64le/ibm )
+    # Darwin power9 - extract version from module environment.
+    string(REPLACE ":" ";" modules $ENV{LOADEDMODULES})
+    foreach( module ${modules} )
+      if( ${module} MATCHES "^gcc" )
+        string( REGEX REPLACE
+          "[^0-9]*([0-9]+).([0-9]+).([0-9]+)" "\\1.\\2.\\3"
+          gcc_version  ${module} )
+      elseif( NOT DEFINED xlc_version AND ${module} MATCHES "^ibm/xlc" )
+        string( REGEX REPLACE
+          "[^0-9]*([0-9]+).([0-9]+).([0-9]+).([0-9]+).*" "\\1.\\2.\\3.\\4"
+          xlc_version ${module} )
+        string( REGEX REPLACE "[^0-9]*([0-9]+).([0-9]+).([0-9]+).*" "\\1.\\2.\\3"
+          xlc_version_3 ${xlc_version} )
+#      Only cuda-10.1 is currently supported
+      elseif( ${module} MATCHES "^cuda" )
+        string( REGEX REPLACE
+          "[^0-9]*([0-9]+).([0-9]+)" "\\1.\\2"
+          cuda_version ${module} )
+        if( NOT ${cuda_version} STREQUAL "10.1" )
+          message( FATAL_ERROR "Only cuda/10.1 is currently supported."
+            " Found cuda/${cuda_version} in your environment. Replace your cuda"
+            " module with cuda/10.1 or unload the cuda module.")
+        endif()
+      endif()
+    endforeach()
+    # Only redhat 7.7 is currently supported
+    # file( READ /etc/redhat-release rhr )
+    #string( REGEX REPLACE "[^0-9]*([0-9]+).([0-9]+).*" "\\1.\\2"
+    #   redhat_version  ${rhr} )
+    set( config_file "/projects/opt/ppc64le/ibm/xlc-${xlc_version}/xlC/${xlc_version_3}/etc/xlc.cfg.rhel.7.7.gcc.${gcc_version}.cuda.10.1" )
+    if( EXISTS ${config_file} )
+      string( APPEND CMAKE_C_FLAGS " -F${config_file}" )
+    else()
+      message( FATAL_ERROR
+        "IBM XLC selected (Darwin), but requested config file was not found."
+        "\nconfig_file = ${config_file}" )
+    endif()
+    unset( gcc_version )
+    unset( xl_version )
+    unset( config_file )
```

Comment:
This is a big mess.  XL needs special configure options to actually support C++14.  Evidently, these options are not universal across platforms.  On rzansel, we needed the `--gcc-toolchain=` option, but that isn't supported on Darwin.  On darwin we must use the `-F config` option.  The path and name of the configuration file depends on the version of XL, gcc, cuda and the OS.  Very ugly.

Reply:
Thanks for the explanation, that is awful!